### ISSUE TITLE
cleos get table lookup fails when specifying an account name that is entirely numerical as lower bound param

### ISSUE THREAD

**Issue by systemzax** [EOSIO/eos#4442]
When calling cleos get table with -L lower parameter, if the key is a string of 12 numerical characters, it will not be used in the lookup and will result in returning the first row as the lower bound of the query scope.

**Comment by systemzax** [comment]
titan@eostitan:~$ cleos get table eosio eosio voters -L 11111111111b --limit 2
{
  "rows": [{
      "owner": "11111111111b",
      "proxy": "",
      "producers": [],
      "staked": 2000,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    },{
      "owner": "111111111122",
      "proxy": "",
      "producers": [],
      "staked": 2000,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    }
  ],
  "more": true
}
titan@eostitan:~$ cleos get table eosio eosio voters -L 111111111122 --limit 2
{
  "rows": [{
      "owner": "111111111114",
      "proxy": "",
      "producers": [],
      "staked": 46200,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    },{
      "owner": "11111111111a",
      "proxy": "",
      "producers": [],
      "staked": 2000,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    }
  ],
  "more": true
}
titan@eostitan:~$

**Comment by systemzax** [comment]
Same issue occurs when using http plugin get_table_rows

**Comment by okayplanet** [comment]
I honestly do not think the lower bound works at all on that table.

```
cleos get table --lower 10 --limit 2 eosio eosio voters
{
  "rows": [{
      "owner": "111111111114",
      "proxy": "",
      "producers": [],
      "staked": 46200,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    },{
      "owner": "11111111111a",
      "proxy": "",
      "producers": [],
      "staked": 2000,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    }
  ],
  "more": true
}
```

Same results with lower bound of 2, or any number.

```
cleos get table --lower 1 --limit 2 eosio eosio voters
{
  "rows": [{
      "owner": "111111111114",
      "proxy": "",
      "producers": [],
      "staked": 46200,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    },{
      "owner": "11111111111a",
      "proxy": "",
      "producers": [],
      "staked": 2000,
      "last_vote_weight": "0.00000000000000000",
      "proxied_vote_weight": "0.00000000000000000",
      "is_proxy": 0
    }
  ],
  "more": true
}
```

**Comment by okayplanet** [comment]
I've confirmed this bug.  Please fix b1.

**Comment by arhag** [comment]
@systemzax:

There is an ambiguity between a `uint64_t` number and an account name consisting of only digits (which is represented by a completely different `uint64_t` number).

There is a hack to resolve such ambiguities: simple add whitespace to the beginning or end of the string containing the account name and you will force the parser to treat it like an account name rather than a raw number.

In other words, if you want to find 2 voters starting with the voter that has the account name "111111111122", you should be able to do so using the cleos command:
```
cleos get table eosio eosio voters -L " 111111111122" --limit 2
```

**Comment by systemzax** [comment]
@arhag Amazing! I've tested this workaround and it works with both cleos and http plugin.

Much appreciated!
